flink-libraries/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliResultView.java

```diff
@@ -150,9 +150,12 @@ protected void openRow() {
 		view.open(); // enter view
 	}
 
-	protected void stopRetrieval() {
+	protected void stopRetrieval(boolean cleanUpQuery) {
 		// stop retrieval
-		refreshThread.isRunning = false;
+		if (refreshThread.isRunning) {
```

Comment:
This change does still not fix the potential concurrency issue that I mentioned. Once `isRunning` is set to `false` the other thread might check for the `cleanUpQuery` value before the next line in this thread is evaluated. A proper implementation looks like:

```
refreshThread.cleanUpQuery = cleanUpQuery;
refreshThread.isRunning = false;
```